commonmark/src/main/java/org/commonmark/internal/InlineParserImpl.java

```diff
@@ -76,6 +76,12 @@ public Scanner scanner() {
         return scanner;
     }
 
+    public InlineParser addInlineParser(Character c, List<InlineContentParser> parsers) {
+        this.inlineParsers.put(c, parsers);
```

Comment:
So this will currently override any existing built-in parsers with that same special character. I think we should do the same thing as for block parsers, see https://github.com/commonmark/commonmark-java/blob/1bf85c6c240bb9a5ab16a931595e83db5526c4e2/commonmark/src/main/java/org/commonmark/parser/Parser.java#L208

So try the custom parsers first. The custom parser can decide not to handle something, in which case the built-in parsers are tried next.